document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -357,17 +362,16 @@ public CloseableIterator<Document> search(Query query) {
     }
 
     String pgSqlQuery = sqlBuilder.toString();
-    try {
-      PreparedStatement preparedStatement =
-          buildPreparedStatement(pgSqlQuery, paramsBuilder.build());
-      LOGGER.debug("Executing search query to PostgresSQL:{}", preparedStatement.toString());
-      ResultSet resultSet = preparedStatement.executeQuery();
+    try (Connection connection = client.getConnection();
+        PreparedStatement preparedStatement =
```

Comment:
Wherever we are returning ClosableIterator, let the client close it. 
Let's say if the client doesn't close it, would the borrowed connection be returned back to the pool after some idle time?